src/Compilers/Core/Portable/InternalUtilities/Hash.cs

```diff
@@ -87,7 +87,7 @@ internal static int CombineValues<T>(T[]? values, int maxItemsToHash = int.MaxVa
                 // Should end up with a constrained virtual call to object.GetHashCode (i.e. avoid boxing where possible).
                 if (value != null)
                 {
-                    hashCode = Hash.Combine(value.GetHashCode(), hashCode);
+                    hashCode = Hash.Combine(EqualityComparer<T>.Default.GetHashCode(value), hashCode);
```

Comment:
yikes.  i thought this issue was fixed by the runtime a long time ago :(

Reply:
➡️ Discussion for this item is in https://github.com/dotnet/roslyn/pull/43060#discussion_r403348616